bin/run

```diff
@@ -81,103 +97,121 @@ with gfile.GFile(model_file, 'rb') as f:
         })
 
 inputs = []
-for i, spec in enumerate(input_spec):
+
+# If data_prep_path is set then extract data-preprocessing
+# graph from the SavedModel. Finally run it on the input-provider
+if config.data_prep_path:
+    data_file_names = list_files_from_dir(config.data_dir)
+
+    # SavedModel with data pre-processing steps
+    with tf.Session(graph=tf.Graph()) as sess:
+        tf.saved_model.loader.load(sess, ['serve'], config.data_prep_path)
+    
+    # Extract graph definition
+    data_prep_gdef = sess.graph_def
+
+    def provide_input():
+            return data_prep_from_saved_model(data_prep_gdef , 
+                data_file_names,
+                config.batch_size,
+                config.data_prep_start_node,
+                config.data_prep_end_node)
+
+    inputs.append(provide_input)
+else:
     def scope(i: int, spec: dict) -> Callable[[], tf.Tensor]:
         def provide_input() -> tf.Tensor:
             pl = tf.placeholder(tf.float32, shape=spec['shape'], name="api/{}".format(i))
             return pl
-
         return provide_input
-
-    inputs.append(scope(i, spec))
-
-
-if config.input_file != '':
-    # TODO: Currently this allows for only one input
-    # If we want to load models with multiple diffrent inputs
-    # We will need to save our inputs as a list of np.array
-    input_data = [np.load(config.input_file)]
-    assert len(input_data) == len(input_spec), 'the input_data length ({}) should be equal the number of inputs required by the loaded model ({})'.format(
-        len(input_data), len(input_spec))
-else:
-    input_data = []
-    for spec in input_spec:
-        input_data.append(
-            np.random.standard_normal(spec['shape']).tolist()
+            
+    inputs = [scope(i, spec) for i, spec in enumerate(input_spec)]
+
+if config.data_prep_path == '':
+    if config.input_file != '':
+        # TODO: Currently this allows for only one input
+        # If we want to load models with multiple different inputs
+        # We will need to save our inputs as a list of np.array
+        input_data = [np.load(config.input_file)]
+        assert len(input_data) == len(input_spec), 'the input_data length ({}) should be equal the number of inputs required by the loaded model ({})'.format(
+            len(input_data), len(input_spec))
+    else:
```

Comment:
There is an extra indent in this code block now

Reply:
Ah my mistake!

Reply:
no problem